backend/test_observer/data_access/models.py

```diff
@@ -296,6 +295,8 @@ class TestResult(Base):
     )
     test_case: Mapped["TestCase"] = relationship()
 
+    historic_results: list = []
+
```

Comment:
Perhaps this property is not needed?

Reply:
This property is actually needed. The reason for this seems to be that the conversion from TestResult -> TestResultDTO happens after we return the response for this endpoint. Because of this, we don't have defined the historic_results property for TestResults and mypy complains about it.

Reply:
Oh I see. Then there should be a way to pass it directly to the pydantic model or perhaps a serialization only field. Since this is not an orm field

Reply:
I was thinking only Mapped fields are considered ORM fields, and the others are serialization-level fields. But, I will look at how we can convert the object to DTO object at request-level.

Reply:
I resolved the issue by converting the test results to TestResultDTO when parsing, and then added the historic results to this field.